### ISSUE TITLE
Maximum number of participants

### ISSUE THREAD

**Issue by four2six** [Loderian/Activities#3]
Hi Mikal,

it would be great in these times to have a maximum number of participants that could be allocated to an event? So it wouldn't be possible to register for an event if the maximum number of participants has been reached. you think that could be possible?

Jan

**Comment by Loderian** [comment]
Yes it should be possible. I've actually been thinking about this, and would be an almost required feature with public join buttons. 
It might take some time though. I probably need a queueing system to prevent, for example, 2 users joining an activity at the same time when its has 15 out of 16 participants. Then for both users the less than 30 participants check passes, but the end result becomes 17/16. So it depends if there are some available queueing systems or I have to make it myself. 
Or there might be another solution, I'll look into it.

**Comment by four2six** [comment]
Thanks! A quick fix would be to sort the participants that have joined by date, and maybe show the join-date in the report. So one could cancel those who joined after the maximum number had been reached.

**Comment by Loderian** [comment]
I ended up with baking the check for participants limit into the (insert participant) sql, so that its all one statement with subqueries. Users who payed with WooCommerce skips the check
